### ISSUE TITLE
Cannot query available issuers via GraphQl

### ISSUE THREAD

**Issue by daiduongvo18** [mollie/magento2#269]
**Describe the bug**
Internal server error occurs when query available issuers.

**Used versions**
- *Magento Version number(eg 2.3.5)*: Magento enterprise 2.3.5-p1
- *Open source/Enterprise/B2b*: Enterprise
- *Mollie version number (Check configuration)*: 1.15.0

**To Reproduce**
1. Create a cart
Request:
```
curl --location --request POST 'http://dev.signify-shop.local/graphql' \
--header 'Content-Type: application/json' \
--header 'Accept: application/json' \
--data-raw '{"query":"mutation {  createEmptyCart }","variables":{}}'
```
Response:
`"data":{"createEmptyCart":"6MNehgMHZaESlj6wtVYEa9h7hIhBwMud"}}`
3. Query issuers
Request:
```
curl --location --request POST 'http://dev.signify-shop.local/graphql' \
--header 'Content-Type: application/json' \
--header 'Accept: application/json' \
--data-raw '{
    "query": "{  cart(cart_id: \\"6MNehgMHZaESlj6wtVYEa9h7hIhBwMud\\") {    id        available_payment_methods {      code      title      mollie_available_issuers {        code        name      }    }      }}",
    "variables": {}
}'
```

**Expected behavior**
A list of available payment method along with theirs issuers is returned.

**Actual behavior**
Internal server error
```
{
    "errors": [
        {
            "debugMessage": "Mollie\\Payment\\GraphQL\\Resolver\\Cart\\AvailableIssuersForMethod cannot function as a GraphQL resolver",
            "message": "Internal server error",
            "extensions": {
                "category": "internal"
            },
            "trace": [
                {
                    "file": "/app/vendor/magento/framework/GraphQl/Schema/Type/Output/ElementMapper/Formatter/Fields.php",
                    "line": 160,
                    "call": "Magento\\Framework\\GraphQl\\Query\\Resolver\\Factory::createByClass('Mollie\\Payment\\GraphQL\\Resolver\\Cart\\AvailableIssuersForMethod')"
                },
                {
                    "file": "/app/vendor/magento/framework/GraphQl/Schema/Type/Output/ElementMapper/Formatter/Fields.php",
                    "line": 98,
                    "call": "Magento\\Framework\\GraphQl\\Schema\\Type\\Output\\ElementMapper\\Formatter\\Fields::getFieldConfig(instance of Magento\\Framework\\GraphQl\\Config\\Element\\Type, GraphQLType: AvailablePaymentMethod, instance of Magento\\Framework\\GraphQl\\Config\\Element\\Field)"
                },
                {
                    "file": "/app/vendor/webonyx/graphql-php/src/Type/Definition/FieldDefinition.php",
                    "line": 90,
                    "call": "Magento\\Framework\\GraphQl\\Schema\\Type\\Output\\ElementMapper\\Formatter\\Fields::Magento\\Framework\\GraphQl\\Schema\\Type\\Output\\ElementMapper\\Formatter\\{closure}()"
                },
                {
                    "file": "/app/vendor/webonyx/graphql-php/src/Type/Definition/ObjectType.php",
                    "line": 151,
                    "call": "GraphQL\\Type\\Definition\\FieldDefinition::defineFieldMap(GraphQLType: AvailablePaymentMethod, instance of Closure)"
                },
                {
                    "file": "/app/vendor/webonyx/graphql-php/src/Validator/Rules/OverlappingFieldsCanBeMerged.php",
                    "line": 255,
                    "call": "GraphQL\\Type\\Definition\\ObjectType::getFields()"
                },
                {
                    "file": "/app/vendor/webonyx/graphql-php/src/Validator/Rules/OverlappingFieldsCanBeMerged.php",
                    "line": 166,
                    "call": "GraphQL\\Validator\\Rules\\OverlappingFieldsCanBeMerged::internalCollectFieldsAndFragmentNames(instance of GraphQL\\Validator\\ValidationContext, GraphQLType: AvailablePaymentMethod, instance of GraphQL\\Language\\AST\\SelectionSetNode, array(0), array(0))"
                },
                {
                    "file": "/app/vendor/webonyx/graphql-php/src/Validator/Rules/OverlappingFieldsCanBeMerged.php",
                    "line": 97,
                    "call": "GraphQL\\Validator\\Rules\\OverlappingFieldsCanBeMerged::getFieldsAndFragmentNames(instance of GraphQL\\Validator\\ValidationContext, GraphQLType: AvailablePaymentMethod, instance of GraphQL\\Language\\AST\\SelectionSetNode)"
                },
                {
                    "file": "/app/vendor/webonyx/graphql-php/src/Validator/Rules/OverlappingFieldsCanBeMerged.php",
                    "line": 65,
                    "call": "GraphQL\\Validator\\Rules\\OverlappingFieldsCanBeMerged::findConflictsWithinSelectionSet(instance of GraphQL\\Validator\\ValidationContext, GraphQLType: AvailablePaymentMethod, instance of GraphQL\\Language\\AST\\SelectionSetNode)"
                },
                {
                    "file": "/app/vendor/webonyx/graphql-php/src/Language/Visitor.php",
                    "line": 410,
                    "call": "GraphQL\\Validator\\Rules\\OverlappingFieldsCanBeMerged::GraphQL\\Validator\\Rules\\{closure}(instance of GraphQL\\Language\\AST\\SelectionSetNode, 'selectionSet', instance of GraphQL\\Language\\AST\\FieldNode, array(9), array(8))"
                },
                {
                    "file": "/app/vendor/webonyx/graphql-php/src/Language/Visitor.php",
                    "line": 466,
                    "call": "GraphQL\\Language\\Visitor::GraphQL\\Language\\{closure}(instance of GraphQL\\Language\\AST\\SelectionSetNode, 'selectionSet', instance of GraphQL\\Language\\AST\\FieldNode, array(9), array(8))"
                },
                {
                    "file": "/app/vendor/webonyx/graphql-php/src/Language/Visitor.php",
                    "line": 273,
                    "call": "GraphQL\\Language\\Visitor::GraphQL\\Language\\{closure}(instance of GraphQL\\Language\\AST\\SelectionSetNode, 'selectionSet', instance of GraphQL\\Language\\AST\\FieldNode, array(9), array(8))"
                },
                {
                    "file": "/app/vendor/webonyx/graphql-php/src/Validator/DocumentValidator.php",
                    "line": 226,
                    "call": "GraphQL\\Language\\Visitor::visit(instance of GraphQL\\Language\\AST\\DocumentNode, array(2))"
                },
                {
                    "file": "/app/vendor/webonyx/graphql-php/src/Validator/DocumentValidator.php",
                    "line": 118,
                    "call": "GraphQL\\Validator\\DocumentValidator::visitUsingRules(instance of Magento\\Framework\\GraphQl\\Schema, instance of GraphQL\\Utils\\TypeInfo, instance of GraphQL\\Language\\AST\\DocumentNode, array(29))"
                },
                {
                    "file": "/app/vendor/webonyx/graphql-php/src/GraphQL.php",
                    "line": 149,
                    "call": "GraphQL\\Validator\\DocumentValidator::validate(instance of Magento\\Framework\\GraphQl\\Schema, instance of GraphQL\\Language\\AST\\DocumentNode, array(29))"
                },
                {
                    "file": "/app/vendor/webonyx/graphql-php/src/GraphQL.php",
                    "line": 90,
                    "call": "GraphQL\\GraphQL::promiseToExecute(instance of GraphQL\\Executor\\Promise\\Adapter\\SyncPromiseAdapter, instance of Magento\\Framework\\GraphQl\\Schema, '{  cart(cart_id: \"6MNehgMHZaESlj6wtVYEa9h7hIhBwMud\") {    id        available_payment_methods {      code      title      mollie_available_issuers {        code        name      }    }      }}', null, instance of Magento\\GraphQl\\Model\\Query\\Context, array(0), null, null, null)"
                },
                {
                    "file": "/app/vendor/magento/framework/GraphQl/Query/QueryProcessor.php",
                    "line": 74,
                    "call": "GraphQL\\GraphQL::executeQuery(instance of Magento\\Framework\\GraphQl\\Schema, '{  cart(cart_id: \"6MNehgMHZaESlj6wtVYEa9h7hIhBwMud\") {    id        available_payment_methods {      code      title      mollie_available_issuers {        code        name      }    }      }}', null, instance of Magento\\GraphQl\\Model\\Query\\Context, array(0), null)"
                },
                {
                    "file": "/app/vendor/magento/module-graph-ql/Controller/GraphQl.php",
                    "line": 160,
                    "call": "Magento\\Framework\\GraphQl\\Query\\QueryProcessor::process(instance of Magento\\Framework\\GraphQl\\Schema, '{  cart(cart_id: \"6MNehgMHZaESlj6wtVYEa9h7hIhBwMud\") {    id        available_payment_methods {      code      title      mollie_available_issuers {        code        name      }    }      }}', instance of Magento\\GraphQl\\Model\\Query\\Context, array(0))"
                },
                {
                    "file": "/app/vendor/magento/framework/Interception/Interceptor.php",
                    "line": 58,
                    "call": "Magento\\GraphQl\\Controller\\GraphQl::dispatch(instance of Magento\\Framework\\App\\Request\\Http)"
                },
                {
                    "file": "/app/vendor/magento/framework/Interception/Interceptor.php",
                    "line": 138,
                    "call": "Magento\\GraphQl\\Controller\\GraphQl\\Interceptor::___callParent('dispatch', array(1))"
                },
                {
                    "file": "/app/vendor/magento/module-page-cache/Model/App/FrontController/BuiltinPlugin.php",
                    "line": 73,
                    "call": "Magento\\GraphQl\\Controller\\GraphQl\\Interceptor::Magento\\Framework\\Interception\\{closure}(instance of Magento\\Framework\\App\\Request\\Http)"
                },
                {
                    "file": "/app/vendor/magento/framework/Interception/Interceptor.php",
                    "line": 135,
                    "call": "Magento\\PageCache\\Model\\App\\FrontController\\BuiltinPlugin::aroundDispatch(instance of Magento\\GraphQl\\Controller\\GraphQl\\Interceptor, instance of Closure, instance of Magento\\Framework\\App\\Request\\Http)"
                },
                {
                    "file": "/app/vendor/magento/framework/Interception/Interceptor.php",
                    "line": 153,
                    "call": "Magento\\GraphQl\\Controller\\GraphQl\\Interceptor::Magento\\Framework\\Interception\\{closure}(instance of Magento\\Framework\\App\\Request\\Http)"
                },
                {
                    "file": "/app/generated/code/Magento/GraphQl/Controller/GraphQl/Interceptor.php",
                    "line": 26,
                    "call": "Magento\\GraphQl\\Controller\\GraphQl\\Interceptor::___callPlugins('dispatch', array(1), array(2))"
                },
                {
                    "file": "/app/vendor/magento/framework/App/Http.php",
                    "line": 116,
                    "call": "Magento\\GraphQl\\Controller\\GraphQl\\Interceptor::dispatch(instance of Magento\\Framework\\App\\Request\\Http)"
                },
                {
                    "file": "/app/generated/code/Magento/Framework/App/Http/Interceptor.php",
                    "line": 24,
                    "call": "Magento\\Framework\\App\\Http::launch()"
                },
                {
                    "file": "/app/vendor/magento/framework/App/Bootstrap.php",
                    "line": 261,
                    "call": "Magento\\Framework\\App\\Http\\Interceptor::launch()"
                },
                {
                    "file": "/app/pub/index.php",
                    "line": 40,
                    "call": "Magento\\Framework\\App\\Bootstrap::run(instance of Magento\\Framework\\App\\Http\\Interceptor)"
                }
            ]
        }
    ]
}
```

**Screenshots**
N/A

**Additional context**
N/A

Thank you for checking.

**Comment by sontran18** [comment]
Hi Team,

Any roadmap for fixing this issue? 
Seems that the GraphQL is supported by the default Module but still having the issue with getting the mollie available issuers.

Thank you!

**Comment by Frank-Magmodules** [comment]
@Hi all,

Thanks for opening this detailed issue, we have this on our schedule, together with some other features, for next week. 
If this is all working out as planned the fix will be released next week.

Thank you for your patience!

**Comment by Frank-Magmodules** [comment]
Hi @daiduongvo18, @sontran18,  

We have pushed an edit for this issue into the [1.15.1-branch](https://github.com/mollie/magento2/tree/1.15.1). 
Can you let us know if this fixes the issue? We can create a new release after confirmation. 

Thank you!

**Comment by Frank-Magmodules** [comment]
Hi @daiduongvo18, @sontran18,
We are happy to share that we've just released the new [1.15.1](https://github.com/mollie/magento2/releases/tag/v1.15.1) version with a fix for this issue. Thank you again for the detailed reports and patience. We are closing this issue now but please feel free to reopen the issue if this still occurs.

**Comment by daiduongvo18** [comment]
Hi @Frank-Magmodules 
I can get available issuers now.
Thank you for your quick action.
